TVShows/TVShows/Modules/Login/LoginPresenter.swift

```diff
@@ -30,67 +39,100 @@ final class LoginPresenter {
         self.interactor = interactor
         self.wireframe = wireframe
     }
-    
 }
 
 // MARK: - Extensions -
 
 extension LoginPresenter: LoginPresenterInterface {
     
-    func isUserInputValid(email: String?, password: String?) -> Bool {
-        guard
-            let email = email,
-            let password = password,
-            email != "" && password != ""
-        else {
-            return false
-        }
-        return true
+    func configure(with output: Login.ViewOutput) -> Login.ViewInput {
+        observablesSetup(with: output)
+        return Login.ViewInput(isInputValid: isInputValid)
     }
+    
+}
 
-    func didSelectLoginAction(with email: String?, password: String?, shouldRemember: Bool) {
-        guard isUserInputValid(email: email, password: password) else { return }
-        let router = Router.login(email: email!, password: password!)
-        interactorLoginRegisterApiCall(with: router, shouldRemember: shouldRemember)
+private extension LoginPresenter {
+    
+    func observablesSetup(with output: Login.ViewOutput) {
+        let credentials = Signal.combineLatest(
+            output.emailChanges.compactMap { $0 }.asSignal(onErrorJustReturn: "") ,
+            output.passwordChanges.compactMap { $0 }.asSignal(onErrorJustReturn: "")
+        )
+        
+        credentials
+            .map(validateInput)
+            .asObservable()
+            .bind(to: isInputValid)
+            .disposed(by: disposeBag)
+
+        loginActionObservableSetup(with: output, credentials: credentials)
+        registerActionObservableSetup(with: output, credentials: credentials)
     }
 
-    func didSelectRegisterAction(with email: String?, password: String?, shouldRemember: Bool) {
-        guard isUserInputValid(email: email, password: password) else { return }
-        let router = Router.register(email: email!, password: password!)
-        interactorLoginRegisterApiCall(with: router, shouldRemember: shouldRemember)
+    func loginActionObservableSetup(
```

Comment:
```suggestion
    func handle(loginAction: Driver<Void>
```
This is how you want the signature of the function to look